src/PKSim.Core/Services/SimulationModelCreator.cs

```diff
@@ -66,20 +66,27 @@ public void CreateModelFor(Simulation simulation, bool shouldValidate = true, bo
 
       private void updateSimulationAfterModelCreation(Simulation simulation)
       {
-         //last step. Once the model has been created, it is necessary to set the id of the simulation 
-         //in all parameter defined in the model
-         _parameterIdUpdater.UpdateSimulationId(simulation);
-
-         //local molecule parameters parameter id need to be updated as well
-         var allParameters = _containerTask.CacheAllChildren<IParameter>(simulation.Model.Root);
+         
          var individual = simulation.Individual;
 
+         var allIndividualParameters = _containerTask.CacheAllChildren<IParameter>(individual);
+         var allSimulationParameters = _containerTask.CacheAllChildren<IParameter>(simulation.Model.Root);
+
+         //Once the model has been created, it is necessary to update origin to ensure that references are set properly
+         //Simulation id for all parameters
+         _parameterIdUpdater.UpdateSimulationId(simulation);
+         //Parameter Ids for all parameters coming from individuals (as some where created in the spatial structure independently from the individual)
+         _parameterIdUpdater.UpdateParameterIds(allIndividualParameters, allSimulationParameters);
```

Comment:
@Yuri05 This is now required to ensure that we still have the connection individual <=> simulation in PKsim.